### PR TITLE
loader, plugin: load decorated callables only

### LINKED ISSUE #859: Module members with attributes named interval, rule, etc. confused for decorated functions
Modules with 'interval' config option don't work, they throw this error:

TypeError: 'ValidatedAttribute' object is not iterable (file "/home/julen/projects/sopel/sopel/bot.py", line 177, in register)

If i change the config option name to something else they work properle

Tryed with this to discard any other problem:

interval.py
import sopel
from sopel.config.types import ValidatedAttribute, StaticSection

class IntervalSection(StaticSection):
    interval = ValidatedAttribute('interval', default='5')

def configure(config):
    config.define_section('test', IntervalSection)
    config.test.configure_setting(
        'interval',
        'interval to work'
    )

def setup(bot=None):
    if not bot:
        return
    bot.config.define_section('test', IntervalSection)

### PR DESCRIPTION
**Pull request by Exirel**
### Description

Fix #859 and that's it.

### Checklist

- [x] I have read [CONTRIBUTING.md](https://github.com/sopel-irc/sopel/blob/master/CONTRIBUTING.md)
- [x] I can and do license this contribution under the EFLv2
- [x] No issues are reported by `make qa` (runs `make quality` and `make test`)
- [x] I have tested the functionality of the things this change touches

### CODE REVIEW

File: sopel/loader.py [hunk 1]
```diff
@@ -234,15 +234,16 @@ def clean_module(module, config):
...
-            elif is_triggerable(obj):
+            elif is_sopel_callable and is_triggerable(obj):
                 clean_callable(obj, config)
                 callables.append(obj)
-            elif hasattr(obj, 'interval'):
+            elif is_sopel_callable and hasattr(obj, 'interval'):
                 clean_callable(obj, config)
                 jobs.append(obj)
-            elif is_url_callback(obj):
+            elif is_sopel_callable and is_url_callback(obj):
```
Comment (dgw) [inline sopel/loader.py:246]: Was there a reason you didn't go with a well-placed `continue` instead of adding `is_sopel_callable and` to every `if`?

```python
    for obj in itervalues(vars(module)):
        if callable(obj):
            is_sopel_callable = getattr(obj, '_sopel_callable', False) is True
            if getattr(obj, '__name__', None == 'shutdown':
                shutdowns.append(obj)
            elif not is_sopel_callable:
                continue  # skip it
            elif is_triggerable(obj):
                # et cetera; I'm not gonna retype everything xD
```

(GitHub won't let me make an easily-applied suggestion because the range includes deleted lines. Sigh.)

I get that `setup` (irrelevant here) and `shutdown` won't be decorated with something that adds the `_sopel_callable` attribute, but a `continue` after checking for the special cases should be fine and avoids re-evaluating the same condition over and over.

**Review by dgw — COMMENTED** [review]
There's one question in line notes. Besides that, I'm also curious if it would be useful to add the `_sopel_callable` attribute to the relevant tests in `test/test_plugin.py`—maybe even some `assert not hasattr()`s for decorators that aren't supposed to add it? Just a thought. 😸

File: sopel/loader.py (hunk 1 above)
Comment (Exirel) [reply sopel/loader.py:246]: Yeah you're right. Why I did that is beyond me.

**Comment by Exirel** [comment]
> maybe even some `assert not hasattr()`s for decorators that aren't supposed to add it?

My initial approach was to test the behaviour of the loader function, more than anything else. After all I decided to add a "protected" attribute (starting with `_`), as to make it not public.

So yeah, I'm still not sure. The behaviour is tested, and I added some corner cases to make sure everything is handled properly. So I feel like it's OK.

File: test/test_loader.py [hunk 2]
```diff
@@ -194,10 +204,23 @@ def test_clean_module(testplugin, tmpconfig):
...
+    assert test_mod.ignored_rate not in jobs
+    assert test_mod.ignored_rate not in shutdowns
+    assert test_mod.ignored_rate not in urls
+    # object with a triggerable attribute are ignored by default
+    assert loader.is_triggerable(test_mod.ignored_obj)
     assert test_mod.ignored_obj not in callables
     assert test_mod.ignored_obj not in jobs
     assert test_mod.ignored_obj not in shutdowns
     assert test_mod.ignored_obj not in urls
+    # trickster function is ignored: it's still not a propre plugin callable
```
Comment (dgw) [inline test/test_loader.py:218]:
```suggestion
    # trickster function is ignored: it's still not a proper plugin callable
```

**Review by dgw — APPROVED** [review]
Just one typo to fix while you're squashing. :shipit:

**Comment by Exirel** [comment]
Done.
